db/db_iter.cc

```diff
@@ -1060,10 +1077,20 @@ bool DBIter::TooManyInternalKeysSkipped(bool increment) {
   return false;
 }
 
-bool DBIter::IsVisible(SequenceNumber sequence) {
+bool DBIter::IsVisible(SequenceNumber sequence, const Slice& ts) {
+  int cmp_ts = timestamp_ub_ != nullptr
+                   ? user_comparator_.CompareTimestamp(ts, *timestamp_ub_)
+                   : 0;
+  if (cmp_ts < 0) {
+    return false;
```

Comment:
If ts in key < read_options.ts, it should be visible?

Reply:
Here, cmp_ts < 0 means key.ts is *after* read_opts.timestamp. The timestamp comparator orders newer key to be "smaller".

Reply:
On second thought, I think the logic may be a little twisted here. Let me change it.